crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -121,6 +121,19 @@ def checks
             upgrade_status.end_step
           end
         end
+      rescue ::Crowbar::Error::StartStepRunningError,
+             ::Crowbar::Error::StartStepOrderError,
+             ::Crowbar::Error::SaveUpgradeStatusError => e
+        raise ::Crowbar::Error::UpgradeError.new(e.message)
+      rescue StandardError => e
+        ::Crowbar::UpgradeStatus.new.end_step(
```

Comment:
Here we should probably do explicit check if the step is running. Maybe it should be in all these exceptions, but in prechecks it's especially needed because it can be run without actually starting the step...